platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/Tipset.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2016-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.tipset;
+
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.system.address.Address;
+import com.swirlds.common.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a slice of the hashgraph, containing one "tip" from each event creator.
+ */
+public class Tipset {
+
+    private final AddressBook addressBook;
+
+    /**
+     * The tip generations, indexed by node index.
+     */
+    private final long[] tips;
+
+    /**
+     * Create an empty tipset.
+     *
+     * @param addressBook the current address book
+     */
+    public Tipset(@NonNull final AddressBook addressBook) {
+        this.addressBook = Objects.requireNonNull(addressBook);
+        tips = new long[addressBook.getSize()];
+
+        // Necessary because we currently start at generation 0, not generation 1.
+        Arrays.fill(tips, -1);
+    }
+
+    /**
+     * Build an empty tipset (i.e. where all generations are -1) using another tipset as a template.
+     *
+     * @param tipset the tipset to use as a template
+     * @return a new empty tipset
+     */
+    private static @NonNull Tipset buildEmptyTipset(@NonNull final Tipset tipset) {
+        return new Tipset(tipset.addressBook);
+    }
+
+    /**
+     * <p>
+     * Merge a list of tipsets together.
+     * </p>
+     *
+     * <p>
+     * The resulting tipset will contain the union of the node IDs of all source tipsets. The generation for each node
```

Comment:
`The resulting tipset will contain the union of the node IDs of all source tipsets.` - this statement is not true. If the first tipset in the list has fewer known nodes IDs than a successive one, the resulting tipset will not contain the union of all node ids. What I think it means to say is that it will contain the union of the highest generations across the node ids.

Reply:
not required, but you could add a check in the existing for-loop over all the tipsets to check that the length of `tips` matches `length` and throw an exception if it doesnt.

Reply:
This is a relic of an old implementation strategy that I have since refactored away. Documentation updated to have just the following:

```
     * The generation for each node ID will be equal to the maximum generation found for that node ID from all source
     * tipsets.
```